src/pm.c

```diff
@@ -257,6 +259,45 @@ static void pmSysBoot(bool enable)
   }
 }
 
+static void delay(int tick) {
+  unsigned int startTick = systickGetTick();
+  while (systickGetTick() - startTick < tick);
+}
+
+void pmDeckctrlDfu(bool enable) {
+  // Set WKUP pin to enter DFU mode on deck controller
+  nrf_gpio_cfg_output(WKUP_PIN);
+  if (enable) {
+    nrf_gpio_pin_set(WKUP_PIN);
+  } else {
+    nrf_gpio_pin_clear(WKUP_PIN);
+  }
+  
+  // Cycle system power to reset the deck controller into DFU or normal mode
+  pmSetState(pmSysOff);
+  while (pmGetState() != pmSysOff) {
+    pmProcess();
+    delay(1);
+  }
+
+  delay(10); // Wait a bit to ensure power is down
```

Comment:
Could there be an issue with the AI-deck that needs more time to reset, or we don't actually care since it should not be mounted nor is it used? Anyway the power down time might have to be longer in case of capacity holding the system alive?